sencha-workspace/packages/slate-cbl/src/view/student/skill/OverviewWindowController.js

```diff
@@ -0,0 +1,101 @@
+/*jslint browser: true, undef: true *//*global Ext,Slate*/
+Ext.define('Slate.cbl.view.student.skill.OverviewWindowController', {
+    extend: 'Ext.app.ViewController',
+    alias: 'controller.slate-cbl-student-skill-overviewwindow',
+    requires: [
+//        'Slate.cbl.API',
+//
+//        'Ext.MessageBox',
+//        'Ext.window.Toast'
+        'Slate.cbl.model.Competency',
+        'Slate.cbl.model.Skill'
+    ],
+
+    config: {
+        id: 'slate-cbl-student-skill-overviewwindow', // workaround for http://www.sencha.com/forum/showthread.php?290043-5.0.1-destroying-a-view-with-ViewController-attached-disables-listen-..-handlers
+        control: {
+            '#': {
+                beforeshow: 'onBeforeWindowShow',
+                demorowclick: 'onDemoRowClick',
+                destroy: 'onDestroy'
+            },
+            'combobox[reference=skillCombo]': {
+                change: 'onSkillChange'
+            }
+        }
+    },
+
+    // workaround for http://www.sencha.com/forum/showthread.php?290043-5.0.1-destroying-a-view-with-ViewController-attached-disables-listen-..-handlers
+    applyId: function(id) {
+        return Ext.id(null, id);
+    },
+
+    onBeforeWindowShow: function(overviewWindow) {
+        var me = this,
+            competencyId = overviewWindow.getCompetency(),
+            student = overviewWindow.getStudent(),
+            skillsCombo = me.lookupReference('skillCombo'),
+            skillStore = skillsCombo.getStore(),
+            skillId = overviewWindow.getSkill();
+
+        if (skillId && competencyId) {
+
+            if(!skillStore.isLoaded()) {
+                Slate.cbl.API.getSkills(competencyId, function(response){
+                    skillStore.loadRawData(response.data);
+
+                    skillsCombo.setValue(skillStore.findRecord('ID', skillId));
+                    me.syncDemonstrationsTable();
+                });
+
+            } else {
+                me.syncDemonstrationsTable();
+            }
+        }
+
+        Ext.on('resize', function() {
```

Comment:
This is not an appropriate place to attach these listeners, as the beforeshow/show events can be fired multiple times in a Component's lifecycle. This would only be appropriate if they were paired with removing the same listeners in a hide event (which might be a good idea for performance if the application made it common to have a lot of this remaining instantiated while hidden).

beforerender/render/afterrender are events that run once in a Component's lifecycle before/after its DOM tree has been built. render/afterrender is typically where you'd want to attach custom listeners to DOM elements within the component.

Since these events are being attached to external, global events though we'll want to be sure that they get removed with this component gets destroyed, so that we're not leaving a bunch of extra listeners running and trying to center a destroyed component.

ExtJS has an easy mechanism for this called "managed listeners", which flips the syntax a bit such that object you're listening to events on is an argument rather than the subject of the function call. So you would do this instead:

```
overviewWindow.mon(Ext.GlobalEvents, 'resize', function() {
    overviewWindow.center();
});

overviewWindow.mon(Ext.getWin(), 'scroll', function() {
    overviewWindow.center();
});
```

This sets up the listener as one managed by `overviewWindow`, which means the listener will automatically be removed when overviewWindow gets destroyed